### DISCUSSION TITLE
Unnormalised the non-dimensional aeroelastic system

### DISCUSSION THREAD

**Discussion by Alanpwu** [ImperialCollegeLondon/sharpy#137] [Q&A] [answered]
If an aeroelastic system is normalised with: 

 'ScalingDict': {'length': 0.5 *c_ref,  'speed': u_inf,  'density': rho} , where u_inf = 1, 

and a discrete-time state-space system (A,B,C,D) is obtained. Is there a way to calculate the dimensional state-vector/state-output (q, q_dot / eta, eta_dot) from the non-dimensional ones? For example at u_inf = 140 at a given normalised time. 

Thank you!

**Comment by ngoiz** [comment, marked as answer]
Hi, 

You need to rescale the structural subsystem with the new scaled time (at your desired velocity) and then couple it with the UVLM. This is done at each velocity. Have a look at Eq 16 in Goizueta, N., Wynn, A., & Palacios, R. (2021). Parametric Krylov-based order reduction of aircraft aeroelastic models. In AIAA SciTech Forum (pp. 1–25). https://doi.org/10.2514/6.2021-1798

This is done within SHARPy in the `AsymptoticStability` postproc (`AsymptoticStability.velocity_analysis()`), have a look there as well and the relevant methods used within that function.

**Reply by Alanpwu** [reply]
What if I want to assemble the aeroelastic system in state-space from at u_inf = 140? (ie t_ref  = b/u_inf = b/140)
Is it simply changing the 'ScalingDict':{'speed' : u_inf = 140}?

**Reply by ngoiz** [reply]
Yes, that would work. You are just defining a different scaling factor
